jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -267,20 +267,26 @@ static Content.Source from(ByteBufferPool.Sized byteBufferPool, InputStream inpu
          */
         static Content.Source from(ByteBufferPool.Sized byteBufferPool, InputStream inputStream, long offset, long length)
         {
+            if (offset > 0 && length != 0)
+            {
+                try
+                {
+                    inputStream.skip(offset - 1);
+                    if (inputStream.read() == -1)
+                        throw new IllegalArgumentException("Offset out of range");
+                }
+                catch (IOException e)
+                {
+                    throw new RuntimeIOException(e);
+                }
+            }
```

Comment:
Maybe move this logic and the length handling below into `InputStreamContentSource`, so it can be used elsewhere.

Reply:
Sure, done.